### PR TITLE
Story opsteam 228

### PR DESCRIPTION
**Pull request by dmatusiewicz**
Adding support for ELB.

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -12,8 +12,8 @@ data "template_file" "assume_role_policy" {
   template = "${file("${path.module}/templates/assume_role_policy.json.tmpl")}"
 }
 
-resource "aws_iam_role_policy_attachment" "iam_role_policy_attachment" {
```
Comment (Zordrak) [inline iam.tf:15]: What's going on here / and why something left commented in any case?

File: sg.tf [hunk 2]
```diff
@@ -29,3 +29,45 @@ resource "aws_security_group_rule" "self_egress" {
...
+  count                    = "${length(var.sg_cb_ingress_rule)}"
+  type                     = "ingress"
+  cidr_blocks              = ["${split(",",lookup(var.sg_cb_ingress_rule[count.index],"cidr_blocks"))}"]
+  protocol                 = "${lookup(var.sg_cb_ingress_rule[count.index],"protocol")}"
+  from_port                = "${lookup(var.sg_cb_ingress_rule[count.index],"from_port")}"
+  to_port                  = "${lookup(var.sg_cb_ingress_rule[count.index],"to_port")}"
+  security_group_id        = "${aws_security_group.security_group.id}"
+}
+
+resource "aws_security_group_rule" "ssg_ingress" {
```
Comment (Zordrak) [inline sg.tf]: Why are we creating security group rules inside the module? Rules should be created at the level above where the module is being called from.

File: variables.tf [hunk 3]
```diff
@@ -43,6 +55,7 @@ variable "availability_zones" {
 variable "aws_region" {
   type        = "string"
   description = "The AWS region"
+  default     = ""
```
Comment (Zordrak) [inline variables.tf]: Why has empty default been added here? Should be a mandatory input.

File: templates/consul_asg.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (Zordrak) [inline templates/consul_asg.sh]: Why is this here? Doesn't look like this has any place being in the generic terraform microservice module(!)

**Review by Zordrak — CHANGES_REQUESTED** [review]
Need some serious review before this is anywhere near pullable.

File: templates/consul_asg.sh (hunk 4 above)
Comment (madpawel) [reply templates/consul_asg.sh]: Fixed, file removed we don't need it here

**Review by Zordrak — APPROVED** [review]

**Pull request merged**